src/credential/issuance/03-start-credential-issuance.ts

```diff
@@ -0,0 +1,336 @@
+  type IdentificationContext,
+} from "@pagopa/io-react-native-wallet";
+import parseUrl from "parse-url";
+import { IdentificationError, ValidationFailed } from "../../utils/errors";
+import { IdentificationResultShape } from "../../utils/identification";
+import { withEphemeralKey } from "../../utils/crypto";
+import { createDPopToken } from "../../utils/dpop";
+import { createPopToken } from "../../utils/pop";
+import { CredentialResponse, TokenResponse } from "./types";
+
+/**
+ * Ensures that the credential type requested is supported by the issuer and contained in the
+ * issuer configuration.
+ * @param issuerConf The issuer configuration
+ * @param credentialType The type of the credential to be requested
+ * @returns The credential definition to be used in the request which includes the format and the type and its type
+ */
+const selectCredentialDefinition = (
+  issuerConf: Out<EvaluateIssuerTrust>["issuerConf"],
+  credentialType: Out<StartFlow>["credentialType"]
+): AuthorizationDetail => {
+  const credential_configurations_supported =
+    issuerConf.openid_credential_issuer.credential_configurations_supported;
+
+  const [result] = Object.keys(credential_configurations_supported)
+    .filter((e) => e.includes(credentialType))
+    .map((e) => ({
+      credential_configuration_id: credentialType,
+      format: credential_configurations_supported[e]!.format,
+      type: "openid_credential" as const,
+    }));
+
+  if (!result) {
+    throw new Error(`No credential support the type '${credentialType}'`);
+  }
+  return result;
+};
+
+/**
+ * Ensures that the response mode requested is supported by the issuer and contained in the issuer configuration.
+ * @param issuerConf The issuer configuration
+ * @param credentialType The type of the credential to be requested
+ * @returns The response mode to be used in the request, "query" for PersonIdentificationData and "form_post.jwt" for all other types.
+ */
+const selectResponseMode = (
+  issuerConf: Out<EvaluateIssuerTrust>["issuerConf"],
+  credentialType: Out<StartFlow>["credentialType"]
+): string => {
+  const responseModeSupported =
+    issuerConf.oauth_authorization_server.response_modes_supported;
+
+  const responseMode =
+    credentialType === "PersonIdentificationData" ? "query" : "form_post.jwt";
+
+  if (!responseModeSupported.includes(responseMode)) {
+    throw new Error(`No response mode support the type '${credentialType}'`);
+  }
+
+  return responseMode;
+};
+
+export type StartCredentialIssuance = (
+  issuerConf: Out<EvaluateIssuerTrust>["issuerConf"],
+  credentialType: Out<StartFlow>["credentialType"],
+  context: {
+    wiaCryptoContext: CryptoContext;
+    credentialCryptoContext: CryptoContext;
+    identificationContext: IdentificationContext;
+    walletInstanceAttestation: string;
+    redirectUri: string;
+    overrideRedirectUri?: string; // temporary parameter to override the redirect uri until we have an actual implementation
```

Comment:
This appears to be no longer used

Reply:
Done in a712c49572f0304c55ce62bee5debbc19c106431, thanks.